training-client.mdx

```diff
@@ -69,7 +71,56 @@ input={
 }
 ```
 
-### Input Times
+#### SQL queries as input
+`offline_query` also takes [spine_sql_query](/api-docs#ChalkClient.offline_query.spine_sql_query) as an alternative
+input parameter for retrieving feature values from your offline data store. 
+
+Spine SQL queries are recommended in the following scenarios:
+
+1. You want to retrieve data for multiple rows at once. Chalk will compute an efficient query plan for loading multiple
+rows of data at once. Chalk will also reuse data between rows where appropriate.
+2. You want to query from Chalk as your offline data store. You can reduce unnecessary back-and-forth requests by having
+Chalk execute the SQL query and handle the result rows directly.
+3. You want to use multiple [feature namespaces](/docs/features#namespacing) in your input. Spine SQL queries accept
+multiple feature namespaces as column names.
+
+`spine_sql_query` expects each column in the result to correspond to a feature. Chalk replaces `__` in column names with
+`.` to form Chalk feature names. For example, `seller__id` corresponds to the `Seller.id` feature.
+
+<TipInfo>
+    The "ts" column is always interpreted as the observation time for the row.
```

Comment:
I am not totally sure what our nomenclature is for this area, but I believe that we treat it as the "query execution time" aka the "now time", the point in time that we want info about.

The distinction is mostly that observed-at is usually an output from resolvers, but the execution time is an input since it determines which entities are included/excluded